packages/react-devtools-shared/src/__tests__/profilingCache-test.js

```diff
@@ -749,4 +754,406 @@ describe('ProfilingCache', () => {
+
+    // Static value so reference doesn't change.
+    const style = {color: 'red'};
+
+    utils.act(() => store.profilerStore.startProfiling());
+    utils.act(() =>
+      legacyRender(
+        <>
+          <ComponentNeedsMemo count={1} />
+          <ComponentShallowMemo count={1} />
+          <ComponentNeedsDeepMemo style={style} />
+          <ComponentDeepMemo style={style} />
+        </>,
+        container,
+      ),
+    );
+
+    // Second render has no changes.
+    utils.act(() =>
+      legacyRender(
+        <>
+          <ComponentNeedsMemo count={1} />
+          <ComponentShallowMemo count={1} />
+          <ComponentNeedsDeepMemo style={style} />
+          <ComponentDeepMemo style={style} />
+        </>,
+        container,
+      ),
+    );
+
+    // Third render has shallow changes only.
+    utils.act(() =>
+      legacyRender(
+        <>
+          <ComponentNeedsMemo count={2} />
+          <ComponentShallowMemo count={2} />
+          <ComponentNeedsDeepMemo style={style} />
+          <ComponentDeepMemo style={style} />
+        </>,
+        container,
+      ),
+    );
+
+    // Fourth render has deep changes only.
+    utils.act(() =>
+      legacyRender(
+        <>
+          <ComponentNeedsMemo count={2} />
+          <ComponentShallowMemo count={2} />
+          <ComponentNeedsDeepMemo style={{color: 'red'}} />
+          <ComponentDeepMemo style={{color: 'red'}} />
+        </>,
+        container,
+      ),
+    );
+
+    utils.act(() => store.profilerStore.stopProfiling());
+
+    const allCommitData = [];
+
+    function Validator({commitIndex, previousCommitDetails, rootID}) {
+      const commitData = store.profilerStore.getCommitData(rootID, commitIndex);
+      if (previousCommitDetails != null) {
+        expect(commitData).toEqual(previousCommitDetails);
+      } else {
+        allCommitData.push(commitData);
+        expect(commitData).toMatchSnapshot(
+          `CommitDetails commitIndex: ${commitIndex}`,
+        );
+      }
+      return null;
+    }
+
+    const rootID = store.roots[0];
+
+    for (let commitIndex = 0; commitIndex < 4; commitIndex++) {
+      utils.act(() => {
+        TestRenderer.create(
+          <Validator
+            commitIndex={commitIndex}
+            previousCommitDetails={null}
+            rootID={rootID}
+          />,
+        );
+      });
+    }
+
+    expect(allCommitData).toHaveLength(4);
+
+    // Export and re-import profile data and make sure it is retained.
+    utils.exportImportHelper(bridge, store);
+
+    for (let commitIndex = 0; commitIndex < 4; commitIndex++) {
+      utils.act(() => {
+        TestRenderer.create(
+          <Validator
+            commitIndex={commitIndex}
+            previousCommitDetails={allCommitData[commitIndex]}
+            rootID={rootID}
+          />,
+        );
+      });
+    }
+  });
+
+  it('should properly detect when a component is missing a shallow/deep memo for state', () => {
```

Comment:
I know you're copying precedent with this test, but I find myself wishing the new tests were written with inline snapshots so they were easier to verify. Comparing test to snapshot data is pretty time consuming. I _think_ I see something unexpected in the snapshot data on first read but it's not obvious without a lot of cross referencing and double checking.

Reply:
Yeah I was copying what was done before and am happy to change. Is there a test that is doing inline snapshots that I can copy?

Reply:
Sure. For example, `packages/react-devtools-shared/src/__tests__/inspectedElement-test.js`

https://github.com/facebook/react/blob/2b77ab26adc2601dc223ede75ff4d19bc19c8684/packages/react-devtools-shared/src/__tests__/inspectedElement-test.js#L172-L213

Reply:
Done, this was much easier to read. I did end up only snapshotting the perfInsights part of the store. Made it much easier to read the test, at the small risk of not detecting side effects.